goss-testing/scripts/check_bios_firmware_versions.sh

```diff
@@ -417,26 +439,49 @@ check_if_bmcs_are_reachable
 # for each BMC
 for i in "${NCN_BMCS[@]}"
 do 
-  
+
+  if [[ "$VENDOR" = *"Marvell"* ]] \
+    || [[ "$VENDOR" = "HP"* ]] \
+    || [[ "$VENDOR" = "Hewlett"* ]]; then
+    if [[ "$i" == *"ncn-m001"* ]]; then
+      # login locally
+      ilorest --nologo login 1>/dev/null
+    else
+      # login to the remote bmc 
+      ilorest --nologo login "$i" 1>/dev/null
+    fi
+
+  fi
+
   check_firmware_version "$i" \
     "$BMC_USERNAME" \
     "$IPMI_PASSWORD"
 
   check_bios_version "$i" \
     "$BMC_USERNAME" \
     "$IPMI_PASSWORD"
-    
+
+  if [[ "$VENDOR" = *"Marvell"* ]] \
+    || [[ "$VENDOR" = "HP"* ]] \
+    || [[ "$VENDOR" = "Hewlett"* ]]; then
+    ilorest --nologo logout 1>/dev/null
+  fi
+
 done
 
 if [[ $BASELINE = Y ]]; then
   /root/bin/bios-baseline.sh --check
```

Comment:
You don't want to skip printing the doc links if this check fails, probably.

```suggestion
  /root/bin/bios-baseline.sh --check || rc=1
```

Reply:
This is actually separate from my script.  I don't want to roll changes to that into this PR.

Reply:
Okay, so your changes are not intended to apply to when this is run from Goss?

When it is run from Goss, then it will go inside this conditional, and as you can see, the check_bios_firmware_versions.sh script will exit purely based on the exit code of the bios-baseline script, regardless of what value the $rc variable has been set to. That's your intention?

Reply:
Fixed after talking with you